Source/WTF/wtf/Assertions.h

```diff
@@ -541,6 +541,19 @@ constexpr bool assertionFailureDueToUnreachableCode = false;
 #endif
 #endif
 
+#ifdef __cplusplus
+namespace WTF {
+template <typename T>
+struct AlwaysFalse {
+    static constexpr bool value = false;
+};
+} // namespace WTF
+#define STATIC_ASSERT_NOT_REACHED_WITH_DEPENDENT_TYPE(T) do { \
```

Comment:
Might be worth adding a comment describing the cases where you can use this. Does this only work inside constexpr branches, or could you also make the body of a function template STATIC_ASSERT_NOT_REACHED, or elsewhere?